R/print_contrib_affil.R

```diff
@@ -41,32 +41,75 @@ print_contrib_affil <- function(infosheet) {
-    tidyr::spread(key = affiliation_type, value = affiliation_no) %>% 
-    dplyr::transmute(contrib = purrr::pmap_chr(list(Names, `Primary affiliation`, `Secondary affiliation`),
-                                               ~ dplyr::if_else(is.na(..3),
-                                                                glue::glue("{..1}^{..2}^"),
-                                                                glue::glue("{..1}^{..2},{..3}^")))) %>% 
-    dplyr::pull(contrib) %>% 
-    glue::glue_collapse(., sep = ", ")
+    tidyr::spread(key = affiliation_type, value = affiliation_no)
+  
+  if (output_format == "rmd") {
+    contrib_print <-
+      contrib_data %>% 
+      dplyr::transmute(contrib = purrr::pmap_chr(list(Names, `Primary affiliation`, `Secondary affiliation`),
+                                                 ~ dplyr::if_else(is.na(..3),
+                                                                  glue::glue("{..1}^{..2}^"),
+                                                                  glue::glue("{..1}^{..2},{..3}^")))) %>% 
+      dplyr::pull(contrib) %>% 
+      glue::glue_collapse(., sep = ", ")
+  } else if (output_format == "html") {
+    contrib_print <-
+      contrib_data %>%
+      dplyr::transmute(contrib = purrr::pmap_chr(list(Names, `Primary affiliation`, `Secondary affiliation`),
+                                                 ~ dplyr::if_else(is.na(..3),
+                                                                  glue::glue("{..1}<sup>{..2}</sup>"),
+                                                                  glue::glue("{..1}<sup>{..2},{..3}</sup>")))) %>% 
+      dplyr::pull(contrib) %>% 
+      glue::glue_collapse(., sep = ", ")
+  } else if (output_format == "raw") {
+    contrib_print <-
+      contrib_data %>% 
+      dplyr::transmute(contrib = purrr::pmap_chr(list(Names, `Primary affiliation`, `Secondary affiliation`),
+                                                 ~ dplyr::if_else(is.na(..3),
+                                                                  glue::glue("{..1}{..2}"),
+                                                                  glue::glue("{..1}{..2},{..3}")))) %>% 
+      dplyr::pull(contrib) %>% 
+      glue::glue_collapse(., sep = ", ")
+  }
   
   # Modify data for printing the affiliations
-  affil_print <- 
+  affil_data <- 
     contrib_affil_data %>% 
     dplyr::select(affiliation_no, affiliation) %>% 
     tidyr::drop_na(affiliation) %>% 
-    dplyr::distinct(affiliation, .keep_all = TRUE) %>% 
-    dplyr::transmute(affil = glue::glue("^{affiliation_no}^{affiliation}")) %>% 
-    dplyr::pull(affil) %>% 
-    glue::glue_collapse(., sep = ", ")
+    dplyr::distinct(affiliation, .keep_all = TRUE)
+  
+  if (output_format == "rmd") {
+    affil_print <-
+      affil_data %>% 
+      dplyr::transmute(affil = glue::glue("^{affiliation_no}^{affiliation}")) %>% 
+      dplyr::pull(affil) %>% 
+      glue::glue_collapse(., sep = ", ")
```

Comment:
Again, to reduce redundancies a little these two lines could be applied after the if-else-block, no?